### PR TITLE
Block Editor: Fix the 'Reset all' bug for the 'ResolutionTool' component

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?

PR fixes a bug in the `ResolutionTool` component that occurs when it fails to rest a value via the "Reset all" action. I've also updated the story to match the editors' current design/behavior (see screencasts).

## Why?
Noticed while working on #68294.

When the consumer doesn't control the `ToolsPanel`, resolution tools might need to implement `resetAllFilter`.

## Testing Instructions
1. Open the Storybook in dev - `npm run storybook:dev`.
2. Navigate to the ResolutionControl story.
3. Change resolution.
4. Reset all values.
5. Confirm that the resolution returns to the default value.

### Testing Instructions for Keyboard
Same.

## Screenshots or screencast 

**Before**

https://github.com/user-attachments/assets/77eb8bd0-e822-4432-b2ba-d59b888a077a

**After**

https://github.com/user-attachments/assets/28c5864a-5e0a-4a21-bd9a-f45d19c831b0

### CODE REVIEW

File: packages/block-editor/src/components/resolution-tool/stories/index.story.js [hunk 1]
```diff
@@ -21,28 +21,55 @@ export default {
...
+export const Default = ( {
+	label,
+	panelId,
+	onChange: onChangeProp,
+	...props
+} ) => {
+	const [ attributes, setAttributes ] = useReducer(
+		( prevState, nextState ) => ( { ...prevState, ...nextState } ),
+		{}
+	);
```
Comment (t-hamano) [inline packages/block-editor/src/components/resolution-tool/stories/index.story.js:33]: Question: Is there any reason not to use `useState`?

**Review by t-hamano — APPROVED** [review]
LGTM!

File: packages/block-editor/src/components/resolution-tool/stories/index.story.js (hunk 1 above)
Comment (Mamaduka) [reply packages/block-editor/src/components/resolution-tool/stories/index.story.js:33]: Seemed more straightforward to implement this partial update pattern via `useReducer`.

**Comment by Mamaduka** [comment]
Thanks for testing, @t-hamano!

**Pull request merged**